x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -500,6 +515,119 @@ export class AlertsClient {
     }
   }
 
+  public async getAlertSummary({
+    gte,
+    lte,
+    featureIds,
+    filter,
+    fixedInterval = '1m',
+    index,
+  }: GetAlertSummaryParams) {
+    try {
+      let indexToUse = index;
+      if (featureIds && !isEmpty(featureIds)) {
+        indexToUse = (
+          (await this.getAuthorizedAlertsIndices(
+            Array.isArray(featureIds) ? featureIds : [featureIds]
+          )) ?? []
+        ).join(',');
+      }
+
+      // first search for the alert by id, then use the alert info to check if user has access to it
```

Comment:
Since we are not returning any hits to explicitly check access, are there any concerns about leaking data? Maybe instead of allowing the index to be passed in, we should only accept feature IDs so that the index can be derived from the feature id?

Reply:
I will remove the index, but I do not think the leaking of the active and recovered will be bad because we still apply the auth filter on the query